src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```
@@ -348,6 +389,78 @@
+      <SonarTemporaryProjectSpecificOutDir>$(ProjectSpecificOutDir).tmp</SonarTemporaryProjectSpecificOutDir>
+    </PropertyGroup>
+    <MoveDirectory SourceDirectory="$(ProjectSpecificOutDir)" DestinationDirectory="$(SonarTemporaryProjectSpecificOutDir)" />
+    <MakeDir Directories="$(ProjectSpecificOutDir)" />
+  </Target>
+
+  <!-- In case of Razor compilation the output files generated during the core compilation step are already moved into a temporary folder.
+       This target renames output directories to their final state and creates the ProjectInfo.xml for them.
+  -->
+  <Target Name="SonarFinishRazorCodeAnalysis" AfterTargets="RazorCoreCompile;InvokeSonarWriteProjectDataRazorCompilation" Condition=" $(SonarTemporaryProjectSpecificOutDir) != '' AND '@(RazorCompile)'!='' ">
+    <PropertyGroup>
+      <RazorSonarProjectSpecificOutDir>$(ProjectSpecificOutDir).Razor</RazorSonarProjectSpecificOutDir>
+      <RazorSonarProjectInfo>$(RazorSonarProjectSpecificOutDir)\ProjectInfo.xml</RazorSonarProjectInfo>
+      <RazorSonarErrorLogExists Condition=" $(RazorSonarErrorLog) != '' AND  $([System.IO.File]::Exists($(RazorSonarErrorLog))) == 'true' ">true</RazorSonarErrorLogExists>
+    </PropertyGroup>
+    <ItemGroup>
+      <RazorSonarReportFilePath Condition= "$(RazorSonarErrorLogExists) == 'true'" Include="$(RazorSonarErrorLog)" />
+      <RazorSonarQubeSetting Include="sonar.$(SQLanguage).roslyn.reportFilePaths" Condition=" @(RazorSonarReportFilePath) != '' ">
+        <!-- Join the paths with |  -->
+        <!-- This delimiter needs to be the same as PropertiesFileGenerator.RoslynReportPathsDelimiter -->
+        <Value>@(RazorSonarReportFilePath->'%(identity)','|')</Value>
+      </RazorSonarQubeSetting>
+      <RazorSonarQubeSetting Include="sonar.$(SQLanguage).analyzer.projectOutPaths" Condition= "$(RazorSonarErrorLogExists) == 'true'">
+        <Value>$(ProjectSpecificOutDir)</Value>
+      </RazorSonarQubeSetting>
+    </ItemGroup>
+    <!-- Record the list of files as an analysis result -->
+    <ItemGroup Condition=" $(AnalysisFilesExist) == 'true' ">
+      <RazorAnalysisResults Include="$(AnalysisFileList)">
+        <Id>FilesToAnalyze</Id>
+      </RazorAnalysisResults>
+    </ItemGroup>
+
+    <!-- Create the ProjectInfo.xml for the Razor part. -->
+    <WriteProjectInfoFile ProjectName="$(MSBuildProjectName)"
+                          FullProjectPath="$(MSBuildProjectFullPath)"
+                          ProjectLanguage="$(Language)"
+                          ProjectGuid="$(ProjectGuid)"
+                          SolutionConfigurationContents="$(CurrentSolutionConfigurationContents)"
+                          IsTest="$(SonarQubeTestProject)"
+                          IsExcluded="$(SonarQubeExclude)"
+                          AnalysisResults="@(RazorAnalysisResults)"
+                          AnalysisSettings="@(RazorSonarQubeSetting)"
+                          OutputFolder="$(ProjectSpecificOutDir)"
```

This is actually confusing because folders aren't where they are supposed to be (yet). This writes Razor content to the ProjectSpecificOutDir that's meant to contain data from the main build.

Let's do `MoveDirectory` first and then invoke this into the correct `Razor` directory.

```suggestion
                          OutputFolder="$(RazorSonarProjectSpecificOutDir)"
```